### DISCUSSION TITLE
Petri Net

### DISCUSSION THREAD

**Discussion by ple1n** [blainehansen/magmide#29] [Ideas]
I was thinking of using petri net to prevent deadlocks. [my post]( https://www.reddit.com/r/rust/comments/168y9sn/petri_net_for_tokio/).

Alternatively I may use actor model which processes messages and does not use locks, but I think sometimes using locks is more efficient and easier to code, which may not be completely dodged with alternative methods. Anyway, if it's possible to prove the absence of deadlocks, expressing a synchronized program with locks should be an equally viable path.

I followed the links in a comment there, and found some really old-looking academic websites. I downloaded the tina tools and it was old-fashioned glitchy GUI with weirdly small font. And it's not opensource.

**Comment by blainehansen** [comment]
I'm not familiar with petri net theory, and I'm sure it could be very useful!

This discussion seems interesting, but I don't think it's extremely relevant to the Magmide project at this early stage. The Magmide design is for a general purpose proof assistant, which would be equally capable of formalizing petri nets as any other structure within the reach of type-theoretical logic. Feel free to share any further advancements or questions you have with the broader internet, I'm not sure I can be very helpful :)
